jxl/src/simd/x86_64/avx.rs

```diff
@@ -123,6 +133,118 @@ impl SimdDescriptor for AvxDescriptor {
+        // enough for these memory operations. We use checked_mul to prevent overflow in offset
+        // calculations before passing to add().
+        unsafe {
+            r0 = _mm256_loadu_ps(input.as_ptr());
+            r1 = _mm256_loadu_ps(input.as_ptr().add(input_stride));
+            r2 = _mm256_loadu_ps(input.as_ptr().add(input_stride.checked_mul(2).unwrap()));
+            r3 = _mm256_loadu_ps(input.as_ptr().add(input_stride.checked_mul(3).unwrap()));
+            r4 = _mm256_loadu_ps(input.as_ptr().add(input_stride.checked_mul(4).unwrap()));
+            r5 = _mm256_loadu_ps(input.as_ptr().add(input_stride.checked_mul(5).unwrap()));
+            r6 = _mm256_loadu_ps(input.as_ptr().add(input_stride.checked_mul(6).unwrap()));
+            r7 = _mm256_loadu_ps(input.as_ptr().add(input_stride.checked_mul(7).unwrap()));
+        }
+
+        // Stage 1: Unpack low/high pairs
+        let t0 = _mm256_unpacklo_ps(r0, r1);
+        let t1 = _mm256_unpackhi_ps(r0, r1);
+        let t2 = _mm256_unpacklo_ps(r2, r3);
+        let t3 = _mm256_unpackhi_ps(r2, r3);
+        let t4 = _mm256_unpacklo_ps(r4, r5);
+        let t5 = _mm256_unpackhi_ps(r4, r5);
+        let t6 = _mm256_unpacklo_ps(r6, r7);
+        let t7 = _mm256_unpackhi_ps(r6, r7);
+
+        // Stage 2: Shuffle to group 32-bit elements
+        let s0 = _mm256_shuffle_ps(t0, t2, 0b01_00_01_00); // _MM_SHUFFLE(1,0,1,0)
+        let s1 = _mm256_shuffle_ps(t0, t2, 0b11_10_11_10); // _MM_SHUFFLE(3,2,3,2)
+        let s2 = _mm256_shuffle_ps(t1, t3, 0b01_00_01_00);
+        let s3 = _mm256_shuffle_ps(t1, t3, 0b11_10_11_10);
+        let s4 = _mm256_shuffle_ps(t4, t6, 0b01_00_01_00);
+        let s5 = _mm256_shuffle_ps(t4, t6, 0b11_10_11_10);
+        let s6 = _mm256_shuffle_ps(t5, t7, 0b01_00_01_00);
+        let s7 = _mm256_shuffle_ps(t5, t7, 0b11_10_11_10);
+
+        // Stage 3: 128-bit permute to finalize transpose
+        let c0 = _mm256_permute2f128_ps(s0, s4, 0x20);
+        let c1 = _mm256_permute2f128_ps(s1, s5, 0x20);
+        let c2 = _mm256_permute2f128_ps(s2, s6, 0x20);
+        let c3 = _mm256_permute2f128_ps(s3, s7, 0x20);
+        let c4 = _mm256_permute2f128_ps(s0, s4, 0x31);
+        let c5 = _mm256_permute2f128_ps(s1, s5, 0x31);
+        let c6 = _mm256_permute2f128_ps(s2, s6, 0x31);
+        let c7 = _mm256_permute2f128_ps(s3, s7, 0x31);
+
+        // SAFETY: The asserts at the top of the function guarantee that the output slice is large
+        // enough for these memory operations. We use checked_mul to prevent overflow in offset
+        // calculations before passing to add().
+        unsafe {
+            _mm256_storeu_ps(output.as_mut_ptr(), c0);
+            _mm256_storeu_ps(output.as_mut_ptr().add(output_stride), c1);
+            _mm256_storeu_ps(
+                output
+                    .as_mut_ptr()
+                    .add(output_stride.checked_mul(2).unwrap()),
```

Comment:
we don't need the checked operations here, we check that there are no overflow above already.